static/js/modules/filters.js

```diff
@@ -1,217 +1,122 @@
+};
+
+Filter.prototype.handleKeydown = function(e) {
+  if (e.which === KEYCODE_ENTER) {
+    e.preventDefault();
+    this.$input.change();
   }
-}
+};
 
-var activateFilter = function(opts) {
-    var $field = $('#category-filters [name=' + opts.name + ']');
-    if ($field.data('temp')) {
-      $field = $('#' + $field.data('temp'));
-    }
-    var $parent = $field.parent();
-    if (opts.value) {
-        $field.val(prepareValue($field, opts.value)).change();
-        $parent.addClass('is-active');
-        ensureVisible($field);
-    } else {
-        $field.val(prepareValue($field, '')).change();
-        $parent.removeClass('is-active');
-    }
+Filter.prototype.handleChange = function() {
+  this.$remove.css('display', this.$input.val() ? 'block' : 'none');
 };
 
-function initCycleFilters() {
-  var cycleSelect = $('.js-cycle');
-  var callbacks = {
-    path: addCyclePath,
-    query: addCycleQuery
-  };
-  cycleSelect.each(function(_, elm) {
-    var $elm = $(elm);
-    var callback = callbacks[$elm.data('cycle-location')];
-    if (callback) {
-      $elm.change(function() {
-        window.location.href = callback($elm.val());
-      });
-    }
-  });
-}
+function DateFilter(elm) {
+  Filter.call(this, elm);
 
-function addCycleQuery(cycle) {
-  return URI(window.location.href)
-    .removeQuery('cycle')
-    .addQuery({cycle: cycle})
-    .toString();
-}
+  this.$minDate = this.$elm.find('.js-min-date');
+  this.$maxDate = this.$elm.find('.js-max-date');
+  this.$elm.on('change', this.handleRadioChange.bind(this));
 
-function addCyclePath(cycle) {
-  var uri = URI(window.location.href);
-  var path = uri.path()
-    .replace(/^\/|\/$/g, '')
-    .split('/')
-    .slice(0, -1)
-    .concat([cycle])
-    .join('/')
-    .concat('/');
-  return uri.path(path).toString();
+  this.fields = ['min_' + this.name, 'max_' + this.name];
 }
 
-function getFields() {
-  return _.chain($('div#filters :input[name]'))
-    .map(function(input) {
-      return $(input).attr('name');
-    })
-    .filter(function(name) {
-      return name && name.slice(0, 1) !== '_';
-    })
-    .uniq()
-    .value();
-}
+DateFilter.prototype = Object.create(Filter.prototype);
+DateFilter.constructor = DateFilter;
 
-var activateInitialFilters = function() {
-    // this activates dropdowns
-    // name filter is activated in the template
-    var open;
-    var qs = URI.parseQuery(window.location.search);
-    var fields = getFields();
-    _.each(fields, function(key) {
-      activateFilter({
-        name: key,
-        value: qs[key]
-      });
-      open = open || qs[key];
-    });
-
-    if (open) {
-      $('body').addClass('is-showing-filters');
-    }
+DateFilter.prototype.handleRadioChange = function(e) {
+  var $input = $(e.target);
+  if (!$input.is(':checked')) { return; }
+  if ($input.attr('data-min-date')) {
+    this.$minDate.val($input.data('min-date'));
+    this.$maxDate.val($input.data('max-date'));
+  }
+  this.$minDate.focus();
```

Comment:
Sure, that's just a holdover from the existing code. We must have wanted it at some point, but I don't remember why.